addons/mrp/models/mrp_production.py

```diff
@@ -1440,6 +1440,8 @@ def action_cancel(self):
         return True
 
     def _action_cancel(self):
+        picking_ids = self.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+        picking_ids.action_cancel()
```

Comment:
Why do you need that. I think it will break the note process since it requires the link between the moves and action_cancel break it

Reply:
It should still happens when we are not in a merge case. We need the warning note

Reply:
@amoyaux 
In which case do We need the warning note?
can you please tell us other use cases?
Thanks.

Reply:
Component MTO with buy route (and multiple receipt steps). Create the MO -> It will create the RFQ -> Validate the RFQ. If you cancel the MO, you will have a warning on the PO for the cancelled quantities

Reply:
I think you should keep the code as before and use a context key to prevent the note in the merge cases instead of removing it everywhere

Reply:
skip_activity already exist

Reply:
I think we can't use `skip_activity` because the current code is first put a note and then cancel the pickings. here -> https://github.com/odoo/odoo/pull/117442/files/a7deb833f70968ca7038f2fa66b9d360a6191257#diff-6f4b1d411b3160a63b8260bf87166f0f20452b9b9fe3f04d28c3734302509f05R1459

Reply:
```
# log an activity on Parent MO if child MO is cancelled.
finish_moves = production.move_finished_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
if finish_moves:
    production._log_downside_manufactured_quantity({finish_move: (production.product_uom_qty, 0.0) for finish_move in finish_moves}, cancel=True)
```

This code does not need if we want to ignore exception notes in pickings that already going to cancel. This is not for `log an activity on Parent MO if child MO is cancelled.`

Reply:
> ```
> # log an activity on Parent MO if child MO is cancelled.
> finish_moves = production.move_finished_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
> if finish_moves:
>     production._log_downside_manufactured_quantity({finish_move: (production.product_uom_qty, 0.0) for finish_move in finish_moves}, cancel=True)
> ```

Also, This only work in 1 step only, 2/3 steps this is not put any note in Parent MO.

Reply:
Here -> https://github.com/odoo/odoo/blob/16.0/addons/mrp/models/mrp_production.py#L1969
we just need to remove this line and the related grouping method because this adds note in pickings that are going to cancel.

Reply:
Just handle the key in the bellow code when you are coming from the merge to skip it then

Reply:
In the merge case, it's working fine I am talking about if we are not in the merge case. Example -> https://drive.google.com/file/d/1V5MnHdpcfBnv4M5X3LvNBBt5WXajnTqB/view?usp=sharing
we need to also prevent this above case.

Reply:
@amoyaux

Reply:
@amoyaux sorry for the late reminder, can you please check the above comment? and let me know the answer.